.github/workflows,.travis.yml: migrate lint and a couple tests to Github Actions

meowsbits opened the pull request:
Travis has a new _Credit_ system that ~~coregeth~~ etclabscore/ is using up (1000 minutes / month) causing Travis to not run anymore until the new month rolls over.

This PR migrates the `make lint` check and a couple simple tests to Github Actions to remove some of the minutes from Travis.

It doesn't touch any Travis jobs that handle deploys (on tags). This is possible in the future, but I just want to get the easy stuff moved over first.

meowsbits committed:
.github/workflows,.travis.yml: migrate lint, test-coregeth@go1.16, test@go1.15 to Actions

Date: 2021-09-21 09:04:04-07:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr commented on .github/workflows/test-linux.yml:
```diff
@@ -0,0 +1,59 @@
+        with:
+          submodules: recursive
+
+      - run: make test
```
Don't we need a `make all` here?

meowsbits replied:
Thanks. Fixed with 4183ae5fae

meowsbits committed:
.github/workflows: add missing 'make all' to test step

re: https://github.com/etclabscore/core-geth/pull/415/files#r713204636

Date: 2021-09-21 09:22:44-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
.github/workflows: only run 'push' action on master; prevent double runs on push+pr branches

Date: 2021-09-21 09:27:47-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
.github/workflows: bump go version (go1.17 latest)

Date: 2021-09-23 07:38:41-07:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
.github/workflows: set ANDROID_HOME to empty to force android_test to skip (like Travis did)

Date: 2021-09-23 07:56:10-07:00
Signed-off-by: meows <b5c6@protonmail.com>
